integration-tests/emulated/src/tests/peregrine/did.rs

```diff
@@ -0,0 +1,114 @@
+	MockNetworkRococo::reset();
+
+	// create the sovereign account of AssetHub
+	let asset_hub_sovereign_account =
+		Peregrine::sovereign_account_id_of(Peregrine::sibling_location_of(AssetHubRococo::para_id()));
+
+	let call: DoubleEncoded<()> = <Peregrine as Parachain>::RuntimeCall::Did(did::Call::create_from_account {
+		authentication_key: did::did_details::DidVerificationKey::Account(asset_hub_sovereign_account.clone()),
+	})
+	.encode()
+	.into();
+
+	let sudo_origin = <AssetHubRococo as Parachain>::RuntimeOrigin::root();
+
+	let parachain_destination: VersionedMultiLocation =
+		ParentThen(Junctions::X1(Junction::Parachain(peregrine::PARA_ID))).into();
+
+	// the Weight parts are copied from logs.
+	let require_weight_at_most = Weight::from_parts(10_000_600_000_000, 200_000_000_000);
+	let origin_kind = OriginKind::Native;
+
+	// XCM is async. In order to know if your call succeeded, we request a response info.
+	let response_info = QueryResponseInfo {
+		destination: ParentThen(Junctions::X1(Junction::Parachain(asset_hub_polkadot::PARA_ID))).into(),
+		max_weight: require_weight_at_most.clone(),
+		// arbitrary query_id
+		query_id: 0,
+	};
+
+	let init_balance = UNIT * 10;
+
+	let xcm = VersionedXcm::from(Xcm(vec![
+		WithdrawAsset((Here, init_balance).into()),
+		BuyExecution {
+			fees: (Here, init_balance).into(),
+			weight_limit: WeightLimit::Unlimited,
+		},
+		Transact {
+			origin_kind,
+			require_weight_at_most,
+			call,
+		},
+		// refund back the withdraw assets.
+		RefundSurplus,
+		// in case of an error, we request the response info.
+		ReportError(response_info),
+	]));
+
+	// give the sovereign account of AssetHub some coins.
+	Peregrine::execute_with(|| {
+		<peregrine_runtime::Balances as Mutate<AccountId>>::set_balance(&asset_hub_sovereign_account, init_balance);
+	});
+
+	//Send XCM message from AssetHub
+	AssetHubRococo::execute_with(|| {
+		assert_ok!(<AssetHubRococo as AssetHubRococoPallet>::PolkadotXcm::send(
+			sudo_origin,
+			Box::new(parachain_destination),
+			Box::new(xcm)
+		));
+
+		type RuntimeEvent = <AssetHubRococo as Parachain>::RuntimeEvent;
+		assert_expected_events!(
+			AssetHubRococo,
+			vec![
+				RuntimeEvent::PolkadotXcm(pallet_xcm::Event::Sent { .. }) => {},
+			]
+		);
+	});
+
+	Peregrine::execute_with(|| {
+		type PeregrineRuntimeEvent = <Peregrine as Parachain>::RuntimeEvent;
+		assert_expected_events!(
+			Peregrine,
+			vec![
+				PeregrineRuntimeEvent::XcmpQueue(cumulus_pallet_xcmp_queue::Event::Success { .. }) => {},
+			]
+		);
+
+		// we also expect that the sovereignAccount of AssetHub has some coins now
+		let balance_after_xcm_call: u128 =
+			<<Peregrine as Parachain>::Balances as Inspect<AccountId>>::balance(&asset_hub_sovereign_account).into();
+
+		assert!(balance_after_xcm_call < init_balance);
+	});
+
+	// No event on the relaychain (message is meant for AssetHub)
+	Rococo::execute_with(|| {
+		assert_eq!(RococoSystem::events().len(), 0);
+	});
```

Comment:
Ideally I would like to check, if the did is really created. But this is not as straight forward as anticipated. Will be checked in the e2e tests with chopsticks. The `reporterror` is blocked by the barrier of assethub.

Reply:
Can't you check on the provider chain state directly? You can capture the result of an `execute_with`.

Reply:
I couldn't find a way to check the state directly, but I was able to produce the Did create event: [bcc4a6a](https://github.com/KILTprotocol/kilt-node/pull/603/commits/bcc4a6a7c1c87ddf75dc47d24c52c26864f80829)

Reply:
That's good. I think it makes sense to check what the value of the event is. I.e., what's the expected DID identifier and what's the expected submitter? They will probably be the same in this case.

Reply:
[30e2dd4](https://github.com/KILTprotocol/kilt-node/pull/603/commits/30e2dd4de4ef862a19f1dbe8cc4cb0c0427e7e84)